### PR TITLE
Allow removal of a single mod in the new loadout editor.

### PR DESCRIPTION
**Pull request by ryan-rushton**
This allows a user to delete a single mod through the new loadout editor without opening the plug picker.

Screen Shot 2022-02-19 at 6 12 25 pm

### CODE REVIEW

File: src/app/loadout/loadout-ui/LoadoutMods.tsx [hunk 1]
```diff
@@ -72,6 +72,15 @@ export default memo(function LoadoutMods({
...
             plug={mod}
+            onClose={
+              onUpdateMods &&
+              (() => {
+                const firstIndex = savedMods.findIndex((savedMod) => savedMod.hash === mod.hash);
+                const newMods = [...savedMods];
+                newMods.splice(firstIndex, 1);
+                onUpdateMods(newMods);
+              })
+            }
```
Comment (ryan-rushton) [inline src/app/loadout/loadout-ui/LoadoutMods.tsx:83]: Do you think it's worth pulling out the single mod being rendered here so we can `useCallback` this? I was a bit undecided.

**Comment by ryan-rushton** [comment]
Already ticked off the last point in #7841

File: src/app/loadout/loadout-ui/LoadoutMods.tsx (hunk 1 above)
Comment (bhollis) [reply src/app/loadout/loadout-ui/LoadoutMods.tsx:83]: Probably?

**Review by bhollis — APPROVED** [review]

File: src/app/loadout/loadout-ui/LoadoutMods.tsx (hunk 1 above)
Comment (ryan-rushton) [reply src/app/loadout/loadout-ui/LoadoutMods.tsx]: Done

**Pull request merged**